post.md

```diff
@@ -0,0 +1,82 @@
+
+Your application's configuration typically dictates the choice between using the [tailwindcss-rails](https://github.com/rails/tailwindcss-rails) gem and the NPM package [tailwindcss](https://github.com/tailwindlabs/tailwindcss).
+
+- **When to use the gem:** opt for the gem when you're focusing on server-side rendering of views, particularly with ERB templates. This approach is streamlined and harmonizes well with the Rails asset pipeline. Typically Rails would be configured to use `importmap` for loading JavaScript packages.
+
+- **When to use the NPM package:** in scenarios where you are rendering views via JavaScript on the client and using Webpack, esbuild, or Rollup to compile your JavaScript and other front end assets. See this [article](https://www.codingzeal.com/post/tailwind-is-a-popular-library-here-is-how-to-add-it-to-redwoodjs) about using Tailwind with RedwoodJS for more details.
+
+## Setting up a new Rails application with the `tailwindcss-rails` gem
+
+To configure a new Rails application with TailwindCSS, use:
+
+```bash
+rails new myapp --css tailwind
+```
+
+## Adding Tailwind to an existing Rails app
+
+For pre-existing Rails applications, add the `tailwindcss-rails` gem to your `Gemfile`:
+
+```ruby
+gem 'tailwindcss-rails'
+```
+
+Then, run:
+
+```bash
+bundle install
+rails tailwindcss:install
+```
+
+Expect the install command to generate the following files:
+
+- `app/assets/stylesheets/application.tailwind.css`
+- `bin/dev`
+- `config/tailwind.config.js`
+- `Procfile.dev`
+
+## What's up with the `bin/dev` script?
+
+The `bin/dev` script helps run development-related tasks concurrently using `foreman`.
+
+The `Procfile.dev` file contains a `tailwindcss:watch` entry. This process watches and compiles the Tailwind utility classes used in your `.erb` files. If this process isn't running, the styles won't render in the browser.
+
+In the [sample application](https://github.com/CodingZeal/dont-bail-on-rails-tailwind), there's a notable change: transitioning from [foreman](https://www.theforeman.org/) to [overmind](https://github.com/DarthSim/overmind). Overmind offers better performance and debugging features than foreman.
+
+## Using scaffold with the Tailwind gem
+
+One of the gems (pun intended!) of the `tailwindcss-rails` is its ability to style scaffolded views using Tailwind's utility classes. This means when you run:
+
+```bash
+rails generate scaffold Post title:string content:text
+```
+
+The views generated will be styled with Tailwind out of the box!
+
+## Custom Tailwind in the sample application
+
+In the sample application, there's a custom Tailwind utility for styling radio buttons. Located in `app/assets/stylesheets/application.tailwind.css`, there's a custom class that makes use of the CSS `:has` pseudo-class for radio button styling. It's important to note that [:has](https://developer.mozilla.org/en-US/docs/Web/CSS/:has#browser_compatibility) is experimental and not supported in all browsers.
```

Comment:
This feels oddly out of place? What's the purpose for calling it out? What does this have to do with "How to add TailwindCSS to Ruby on Rails"?

Reply:
Good call! Removing it from this post. We may write a separate post about this 🤷‍♂️